storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+                                                       'value': 'T%'}},
+                                       {'conjunction': 'or',
+                                        'complement': [{'attribute': 'size',
+                                                        'relation': '<',
+                                                        'value': 1000},
+                                                       {'attribute': 'size',
+                                                        'relation': '>',
+                                                        'value': 512}]}]}}}
+            return {'ArraySearchOverPackagesExample': array_search_example,
+                    'ObjectSearchOverPackagesExample': object_search_example}
+
+    def _set_search_paths(self, paths, resource_name, resource_cls, pk_patt,
+                          rsrc_collection_name):
+        for action in ('search_post', 'new_search'):
+            http_method = {'search_post': 'post', 'new_search': 'get'}.get(
+                action, 'search')
+            operation_id = _get_operation_id(action, resource_name)
+            modifiers = {'search_post': ['search'],
+                         'new_search': ['new_search']}.get(
+                             action, [])
+            path = get_collection_targeting_openapi_path(
+                rsrc_collection_name, modifiers=modifiers)
+            path_dict = paths.setdefault(path, {})
+            responses_meth = '_get_{}_responses'.format(action)
+            parameters_meth = '_get_{}_parameters'.format(action)
+            request_body_meth = '_get_{}_request_body'.format(action)
+            request_body_examples_meth = '_get_{}_request_body_examples'.format(
+                action)
+            parameters = getattr(self, parameters_meth, lambda x: None)(
+                resource_name)
+            request_body_examples = getattr(
+                self, request_body_examples_meth, lambda x: None)(resource_name)
+            request_body_meth = getattr(self, request_body_meth, None)
+            request_body = None
+            if request_body_meth:
+                request_body = request_body_meth(
+                    resource_name, examples=request_body_examples)
+            summary, description = _summarize(
+                action, resource_name, rsrc_collection_name)
+            path_dict[http_method] = OrderedDict([
+                ('summary', summary),
+                ('description', description),
+                ('operationId', operation_id),
+            ])
+            if parameters:
+                path_dict[http_method]['parameters'] = parameters
+            if request_body:
+                path_dict[http_method]['requestBody'] = request_body
+            path_dict[http_method]['responses'] = getattr(
+                self, responses_meth)(resource_name)
+            path_dict[http_method]['tags'] = [rsrc_collection_name]
```

Comment:
It should be possible to remove some of the duplication here with `_set_crud_paths`.